Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Booking.hs

```diff
@@ -16,26 +17,39 @@
 module Storage.Queries.Booking where
 
 import Domain.Types.Booking
+import qualified Domain.Types.Booking.BookingLocation as DBBL
 import Domain.Types.DriverQuote as DDQ
+import qualified Domain.Types.Location as DL
+import qualified Domain.Types.LocationMapping as DLM
 import Domain.Types.Merchant
 import Domain.Types.RiderDetails (RiderDetails)
 import qualified Domain.Types.SearchTry as DST
+import EulerHS.Prelude (whenNothingM_)
 import Kernel.Beam.Functions
 import Kernel.Prelude
 import Kernel.Types.Error
 import Kernel.Types.Id
 import Kernel.Utils.Common
 import qualified Sequelize as Se
+import qualified SharedLogic.LocationMapping as SLM
 import qualified Storage.Beam.Booking as BeamB
 import qualified Storage.Queries.Booking.BookingLocation as QBBL
 import qualified Storage.Queries.DriverQuote as QDQuote
 import qualified Storage.Queries.FareParameters as QueriesFP
+import qualified Storage.Queries.Location as QL
+import qualified Storage.Queries.LocationMapping as QLM
 
 createBooking :: MonadFlow m => Booking -> m ()
 createBooking = createWithKV
 
 create :: MonadFlow m => Booking -> m ()
-create dBooking = QBBL.create dBooking.fromLocation >> QBBL.create dBooking.toLocation >> createBooking dBooking
+create dBooking = do
+  _ <- whenNothingM_ (QL.findById dBooking.fromLocation.id) $ do QL.create dBooking.fromLocation
```

Comment:
Why find first? It will not be there, right?

Reply:
there is one to many mapping for fromLocationId and toLocationId as these are same for searchReq , booking , ride so they will be there if created at the starting